### PR TITLE
[backend] fix contextual team while duplicating scenario #1303

### PR DESCRIPTION
**Pull request by MarineLeM**
- Create new contextual team while duplicating scenario

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/utils/CopyObjectListUtils.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import io.openbas.database.model.Base;
+import jakarta.validation.constraints.NotNull;
+import org.apache.commons.beanutils.BeanUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+
+public class CopyObjectListUtils {
+
+    public static <T extends Base> List<T> copy(@NotNull final List<T> origins, Class<T> clazz) {
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/utils/CopyObjectListUtils.java:12]: Nice !

File: openbas-api/src/test/java/io/openbas/service/ScenarioServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                        scenarioDuplicated.getTeams().stream().filter(team -> team.getContextual().equals(true)).findFirst().orElse(new Team()).getId(),
+                        injectTeam.getId()
+                );
+            } else {
+                assertEquals(noContextualTeam.getId(), injectTeam.getId());
+            }
+        });
+    }
+
+    private Team createTeam(@NotBlank String name, Boolean isContextualTeam){
```
Comment (RomuDeuxfois) [inline openbas-api/src/test/java/io/openbas/service/ScenarioServiceTest.java:108]: You can add this in TeamFixture

**Comment by RomuDeuxfois** [comment]
Test OK for Scenario.
We have the bug in Simulation, it would be great to fix it in this PR too.

File: openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java [hunk 3]
```diff
@@ -51,7 +51,7 @@ public interface TeamRepository extends CrudRepository<Team, String>,
 
   @Query(value="SELECT teams.team_id, teams.team_name, teams.team_description, teams.team_created_at, teams.team_updated_at, teams.team_organization, " +
           "       team_contextual, " +
-          "       coalesce(array_agg(DISTINCT teams_tags.team_id) FILTER ( WHERE teams_tags.team_id IS NOT NULL ), '{}') as team_tags, " +
+          "       coalesce(array_agg(DISTINCT teams_tags.tag_id) FILTER ( WHERE teams_tags.tag_id IS NOT NULL ), '{}') as team_tags, " +
```
[the remark this answers is not in the document]
Comment (MarineLeM) [reply openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java:54]: Before the correction, the request returned the id of the team but what we want is the ids of the tags

File: openbas-api/src/test/java/io/openbas/service/ScenarioServiceTest.java (hunk 2 above)
Comment (MarineLeM) [reply openbas-api/src/test/java/io/openbas/service/ScenarioServiceTest.java:108]: yes my bad i saw the utils/fixtures folder only this morning i will correct that

File: openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (RomuDeuxfois) [reply openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java:54]: Ah I understand, thank you

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**